Mathlib/Algebra/Order/Hom/Monoid.lean

```diff
@@ -453,36 +451,40 @@ end Preorder
 
 section Mul
 
-variable [OrderedCommMonoid α] [OrderedCommMonoid β] [OrderedCommMonoid γ]
+variable [CommMonoid α] [PartialOrder α]
+  [CommMonoid β] [PartialOrder β]
+  [CommMonoid γ] [PartialOrder γ]
 
 /-- For two ordered monoid morphisms `f` and `g`, their product is the ordered monoid morphism
 sending `a` to `f a * g a`. -/
 @[to_additive "For two ordered additive monoid morphisms `f` and `g`, their product is the ordered
 additive monoid morphism sending `a` to `f a + g a`."]
-instance : Mul (α →*o β) :=
+instance [IsOrderedMonoid β] : Mul (α →*o β) :=
   ⟨fun f g => { (f * g : α →* β) with monotone' := f.monotone'.mul' g.monotone' }⟩
 
 @[to_additive (attr := simp)]
-theorem coe_mul (f g : α →*o β) : ⇑(f * g) = f * g :=
+theorem coe_mul [IsOrderedMonoid β] (f g : α →*o β) : ⇑(f * g) = f * g :=
   rfl
 
 @[to_additive (attr := simp)]
-theorem mul_apply (f g : α →*o β) (a : α) : (f * g) a = f a * g a :=
+theorem mul_apply [IsOrderedMonoid β] (f g : α →*o β) (a : α) : (f * g) a = f a * g a :=
   rfl
 
 @[to_additive]
-theorem mul_comp (g₁ g₂ : β →*o γ) (f : α →*o β) : (g₁ * g₂).comp f = g₁.comp f * g₂.comp f :=
+theorem mul_comp [IsOrderedMonoid γ] (g₁ g₂ : β →*o γ) (f : α →*o β) :
+    (g₁ * g₂).comp f = g₁.comp f * g₂.comp f :=
   rfl
 
 @[to_additive]
-theorem comp_mul (g : β →*o γ) (f₁ f₂ : α →*o β) : g.comp (f₁ * f₂) = g.comp f₁ * g.comp f₂ :=
+theorem comp_mul [IsOrderedMonoid β] [IsOrderedMonoid γ] (g : β →*o γ) (f₁ f₂ : α →*o β) :
+    g.comp (f₁ * f₂) = g.comp f₁ * g.comp f₂ :=
   ext fun _ => map_mul g _ _
 
 end Mul
 
 section OrderedCommMonoid
 
-variable {hα : OrderedCommMonoid α} {hβ : OrderedCommMonoid β}
+variable {_ : Preorder α} {_ : Preorder β} {_ : MulOneClass α} {_ : MulOneClass β}
```

Comment:
Not for this PR, but there should be a systematic review of this pattern at some point.